Spring-DI-extension fails when using custom stereotype in Spring @Configuration class

Spring-DI supports custom stereotypes in bean producing method declared in a Spring Configuration class.

``` 
@Component 
@Target({ ElementType.TYPE, ElementType.METHOD }) 
@Retention(RetentionPolicy.RUNTIME)
@Scope("prototype")
public @interface CustomPrototype {}
```

```
@Configuration
public class Config {

@Bean
@CustomPrototype 
public CustomBean customPrototypeBean() {
  // ... 
}

}
```

**Actual behavior**
When having a custom scope as described above spring-di extension fails with an error 

```
Caused by: java.lang.IllegalArgumentException: Not a class
	at org.jboss.jandex.MethodInfo.asClass(MethodInfo.java:349)
	at io.quarkus.spring.di.deployment.SpringDIProcessor.lambda$beanTransformer$0(SpringDIProcessor.java:101)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:174)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at io.quarkus.spring.di.deployment.SpringDIProcessor.beanTransformer(SpringDIProcessor.java:102)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.quarkus.deployment.ExtensionLoader$1.execute(ExtensionLoader.java:941)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:415)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
	at java.lang.Thread.run(Thread.java:748)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
```

**Expected behavior**
Spring-di extension do not fail and gently maps the custom spring stereotype to a CDI stereotype.


**To Reproduce**
Steps to reproduce the behavior:
1. Modify spring-di integration test and add a custom stereotype definition and use it in a @Configuraion class

**Environment**
 - Quarkus version or git rev: 1.0.0.Final and probably above

--- a/extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java
+++ b/extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java
@@ -129,7 +129,8 @@ public class SpringDIProcessor {
         for (final DotName name : stereotypeScopes.keySet()) {
             instances.put(name, index.getAnnotations(name)
                     .stream()
-                    .filter(it -> isAnnotation(it.target().asClass().flags()))
+                    .filter(it -> it.target().kind() == AnnotationTarget.Kind.CLASS
+                            && isAnnotation(it.target().asClass().flags()))
                     .collect(Collectors.toSet()));
         }
         additionalStereotypeBuildItemBuildProducer.produce(new AdditionalStereotypeBuildItem(instances));

--- a/integration-tests/spring-di/src/main/java/io/quarkus/it/spring/AppConfiguration.java
+++ b/integration-tests/spring-di/src/main/java/io/quarkus/it/spring/AppConfiguration.java
@@ -33,11 +33,21 @@ public class AppConfiguration {
         return new AnotherRequestBean();
     }
 
+    @Bean
+    @CustomPrototype
+    public CustomPrototypeBean beanWithCustomPrototype() {
+        return new CustomPrototypeBean();
+    }
+
     private static class SingletonBean {
 
     }
 
     private static class AnotherRequestBean {
+
+    }
+
+    public static class CustomPrototypeBean {
 
     }
 }

--- a/integration-tests/spring-di/src/main/java/io/quarkus/it/spring/CustomPrototype.java
+++ b/integration-tests/spring-di/src/main/java/io/quarkus/it/spring/CustomPrototype.java
@@ -1,0 +1,16 @@
+package io.quarkus.it.spring;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component
+@Target({ ElementType.TYPE, ElementType.METHOD })
+@Retention(RetentionPolicy.RUNTIME)
+@Scope("prototype")
+public @interface CustomPrototype {
+}

--- a/integration-tests/spring-di/src/main/java/io/quarkus/it/spring/InjectedSpringBeansResource.java
+++ b/integration-tests/spring-di/src/main/java/io/quarkus/it/spring/InjectedSpringBeansResource.java
@@ -1,5 +1,7 @@
 package io.quarkus.it.spring;
 
+import static io.quarkus.it.spring.AppConfiguration.CustomPrototypeBean;
+
 import javax.inject.Inject;
 import javax.servlet.http.HttpServletRequest;
 import javax.ws.rs.GET;
@@ -18,6 +20,8 @@ public class InjectedSpringBeansResource {
     RequestBean requestBean;
     @Inject
     SessionBean sessionBean;
+    @Inject
+    CustomPrototypeBean anotherRequestBean;
 
     @GET
     @Produces(MediaType.TEXT_PLAIN)

